src/Featurizers/Structs.h

```diff
@@ -95,6 +95,65 @@ class SingleValueSparseVectorEncoding {
     bool operator!=(SingleValueSparseVectorEncoding const &other) const;
 };
 
+
+/////////////////////////////////////////////////////////////////////////
+
+///  \class        CSREncoding
+///  \brief         Struct that contains information that can be used to
+///                 create a sparse matrix without creating the matrix
+///                 itself. We are using Compressed Sparse Row form
+///                 in this case.
+///
+///                 CSREncoding contains three vectors to express a
+///                 sparse matrix:
+///                 V contains all non-zero element values,
+///                 ColIndex contains column indices of non-zero values,
+///                 RowIndex has one element per row in the matrix and encodes
+///                 the index in V where the given row starts
+///
+
+template <typename T>
+class CSREncoding {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using value_type                        = T;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+
+    std::vector<value_type>          Values;
+    std::vector<std::uint64_t>       ColIndex;
+    std::vector<std::uint64_t>       RowIndex;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    CSREncoding();

```

Comment:
>CSREncoding(); [](start = 4, length = 14)

Remove this default constructor, as it creates the possibility of creating a CSREncoding object that doesn't have all of the values set. For example, is it valid to have a CSREncoding object with values but no ColIndex or RowIndexes? If this is to handle the creation of an empty matrix, create a static member named something like "CreateEmptyMatrix" that passes in empty vectors for Values, ColIndex, and RowIndex.

Reply:
Eigen library can deal with compressed format so CSREncoding is also no longer necessary